solver/llbsolver/solver.go

```diff
@@ -173,12 +179,20 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 			}
 			inp.Ref = workerRef.ImmutableRef
 
-			dt, err := inlineCache(ctx, exp.CacheExporter, r, session.NewGroup(sessionID))
+			dtbi, err := patchBuildInfo(ctx, res, inp.Metadata[exptypes.ExporterImageConfigKey])
```

Comment:
`mergeBuildInfo` `combineBuildInfo`. Another way would be to add llb buildinfo  directly to `inp` and let let exporter to fix it.

Reply:
I was also wondering where in the workflow this was most effective. Doing it in the solver allows me to work directly with the image config and pass the result to the `exporterResponse` in the same place without additions to the exporter. Let me know if you think it's more legitimate to place logic in the exporter.